File: oop/heritage/execice 3/src/Dictionnaire.java
package EX3;

public class Dictionnaire extends Document{
	private String langue;
	private int nbArticles;
	public Dictionnaire() {
		super();
		this.langue="";
		this.nbArticles=0;
	}
	public Dictionnaire(int num, String titre,String langue, int nbArticles) {
		super(num, titre);
		this.langue=langue;
		this.nbArticles=nbArticles;
	}
	public String getLangue() {
		return langue;
	}
	public void setLangue(String langue) {
		this.langue = langue;
	}
	public int getNbArticles() {
		return nbArticles;
	}
	public void setNbArticles(int nbArticles) {
		this.nbArticles = nbArticles;
	}
	@Override
	public String toString() {
		return "Dictionnaire [langue=" + langue + ", nbArticles=" + nbArticles + ", getLangue()=" + getLangue()
				+ ", getNbArticles()=" + getNbArticles() + ", getNumEnregistrement()=" + getNumEnregistrement()
				+ ", getTitre()=" + getTitre() + ", getClass()=" + getClass() + ", hashCode()=" + hashCode()
				+ ", toString()=" + super.toString() + "]";
	}
	
	
}

File: oop/heritage/CPP/exercice3/src/Dictionnaire.cpp
#include "Dictionnaire.hpp"

Dictionnaire::Dictionnaire() : Document(), langue(""), nbArticles(0) {}

Dictionnaire::Dictionnaire(int num, const std::string& titre, const std::string& langue, int nbArticles)
    : Document(num, titre), langue(langue), nbArticles(nbArticles) {}

const std::string& Dictionnaire::getLangue() const {
    return langue;
}

void Dictionnaire::setLangue(const std::string& newLangue) {
    langue = newLangue;
}

int Dictionnaire::getNbArticles() const {
    return nbArticles;
}

void Dictionnaire::setNbArticles(int newNbArticles) {
    nbArticles = newNbArticles;
}

std::string Dictionnaire::toString() const {
    return "Dictionnaire [langue=" + langue + ", nbArticles=" + std::to_string(nbArticles) + ", base=" + Document::toString() + "]";
}
